contracts/EscrowedNori.sol

```diff
@@ -0,0 +1,725 @@
+   */
+  function batchGetEscrowSchedules(address[] calldata accounts)
+    public
+    view
+    returns (EscrowScheduleDetail[] memory)
+  {
+    EscrowScheduleDetail[]
+      memory escrowScheduleDetails = new EscrowScheduleDetail[](
+        accounts.length
+      );
+    for (uint256 i = 0; i < accounts.length; i++) {
+      uint256[] memory escrowScheduleIds = _addressToEscrowScheduleIds[
+        accounts[i]
+      ];
+      for (uint256 j = 0; j < escrowScheduleIds.length; j++) {
+        escrowScheduleDetails[i] = getEscrowSchedule(
+          accounts[i],
+          escrowScheduleIds[j]
+        );
+      }
+    }
+    return escrowScheduleDetails;
+  }
+
+  /**
+   * @notice Returns all governing settings for an escrow schedule.
+   */
+  function getEscrowSchedule(address account, uint256 escrowScheduleId)
+    public
+    view
+    returns (EscrowScheduleDetail memory)
+  {
+    EscrowSchedule storage escrowAcount = _addressToEscrowSchedules[account][
+      escrowScheduleId
+    ];
+    return
+      EscrowScheduleDetail(
+        escrowAcount.currentAmount,
+        account,
+        escrowAcount.startTime,
+        escrowAcount.endTime,
+        escrowAcount.claimedAmount,
+        escrowAcount.totalQuantityRevoked,
+        escrowAcount.exists
+      );
+  }
+
+  /**
+   * @notice Released balance less any claimed amount at current block timestamp.
+   */
+  function claimableBalanceOf(address account) public view returns (uint256) {
+    return _claimableBalanceOf(account, block.timestamp); // solhint-disable-line not-rely-on-time, this is time-dependent
+  }
+
+  /**
+   * @notice Overridden standard ERC777.burn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it so that we can revert.
+   */
+  function burn(uint256, bytes memory) public pure override {
+    revert("eNORI: burning not supported");
+  }
+
+  /**
+   * @notice Overridden standard ERC777.operatorBurn that will always revert
+   *
+   * @dev This function is not currently supported from external callers so we override it so that we can revert.
+   */
+  function operatorBurn(
+    address,
+    uint256,
+    bytes memory,
+    bytes memory
+  ) public pure override {
+    revert("eNORI: burning not supported");
+  }
+
+  /**
+   * @notice Wraps minting of wrapper token and escrow schedule setup.
+   *
+   * @dev If `startTime` is zero no escrow schedule is set up.
+   * Satisfies situations where funding of the account happens over time.
+   *
+   * @param amount uint256 Quantity of `_bridgedPolygonNori` to deposit
+   * @param userData bytes extra information provided by the user (if any)
+   * @param operatorData bytes extra information provided by the operator (if any)
+   */
+  function _depositFor(
+    uint256 removalId,
+    uint256 amount,
+    bytes calldata userData,
+    bytes calldata operatorData
+  ) internal returns (bool) {
+    address recipient = removalId.supplierAddress();
```

Comment:
is there any world where recipient would not be determined by the removal id?

Reply:
this is actually a really interesting thought... the only use case that I can think of for this is if we had minted removals for a supplier and then their ethereum address became compromised and it was important that their sale proceeds no longer went to that address.  However, since the Market contract also decodes the ID to figure out who to pay, I feel like this scenario would be better handled by an intervention on behalf of the supplier where we delist all of their supply, burn it, and re-issue their removals to their new ethereum address.  In fact, this is probably a functionality that we should make sure we have built before launch to protect suppliers who may have a compromised address.

Reply:
I wonder if this is already functionality we have based on the OZ implementations we inherit... we should maybe look into this and make a ticket if there's anything we still need to do

Reply:
Had a discussion with Scott and Jaycen, and came up with this ticket for this issue: https://norinauts.atlassian.net/browse/NO-1805